ecosystem/platform/server/app/javascript/controllers/claim_nft_controller.ts

```diff
@@ -39,24 +43,29 @@ export default class extends Controller<HTMLAnchorElement> {
   }
 
   async submitTransaction(claimDetails: ClaimDetails) {
-    // TODO: Adjust transaction payload to match module expectations.
+    // Module expects signature as array of bytes.
+    const signature = fromHexString(claimDetails.signature.substring(2));
     const transaction = {
       type: 'entry_function_payload',
       function: claimDetails.module_address + '::claim_mint',
       arguments: [
         claimDetails.message,
-        claimDetails.signature,
+        signature,
       ],
       type_arguments: [],
     };
 
     if (claimDetails.wallet_name === 'petra') {
       const pendingTransaction = await window.aptos!.signAndSubmitTransaction(transaction);
-      // TODO: Do something with the transaction.
+      if ('hash' in pendingTransaction && typeof pendingTransaction.hash === 'string') {
+        // TODO: Do something more intelligent with the transaction.
+        window.open(`https://explorer.devnet.aptos.dev/txn/${pendingTransaction.hash}`);
```

Comment:
we need to add the network to the end of this, but other than that... I don't hate the approach ;-)